### PR TITLE
Finished auth endpoints

### PR DESCRIPTION
**Pull request by pingrishabh**
JWT Signing
POST /v1/auth/login
POST /v1/auth/register

### CODE REVIEW

**Review by flozender — APPROVED** [review]

File: server/api/services/auth.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        else reject({ code: 401, message: "Incorrect credentials." });
+      });
+    });
+  }
+
+  static registerUser(email: string, password: string) {
+    return new Promise(async (resolve, reject) => {
+      const user = await User.findOne({ email: email });
+      if (!user) {
+        let newUser = User({ email: email, password: password });
```
Comment (utkarsh-dixit) [inline server/api/services/auth.ts]: Lets `md5` or encrypt the password before storing in the database to ensure the security of users.

File: server/api/services/auth.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import * as mongoose from "mongoose";
+import * as jwt from "jsonwebtoken";
+import { UserModel } from "../models/user";
+
+const User = mongoose.model("User", UserModel);
+
+export class AuthService {
+  static loginUser(email: string, password: string) {
+    return new Promise((resolve, reject) => {
+      User.findOne({ email: email, password: password }, (err, user) => {
```
Comment (utkarsh-dixit) [inline server/api/services/auth.ts]: Same as above

**Review by utkarsh-dixit — CHANGES_REQUESTED** [review]
LGTM!! except two small changes

File: server/api/services/auth.ts (hunk 1 above)
Comment (pingrishabh) [reply server/api/services/auth.ts]: Makes sense. On it. Using bcrypt like discussed.

**Review by utkarsh-dixit — APPROVED** [review]

**Pull request merged**